pkg/apis/certificates/types.go

```diff
@@ -68,6 +68,11 @@ type CertificateSigningRequestSpec struct {
 	//  6. Whether or not requests for CA certificates are allowed.
 	SignerName string
 
+	// durationHint is a hint to the signer in regards to when the issued certificate should expire.
+	// The signer may or may not honor this field.  The well-known kubernetes signers will honor this field
+	// as long as the requested duration is not later than the maximum duration they will honor.
+	DurationHint *metav1.Duration
```

Comment:
I wounder if something like `RequestedValidityDuration` or `RequestedTTL` would be clearer.

Reply:
I just went with `durationHint` because of what I recalled from the SIG meetings.

Reply:
per [conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#units), include the unit in the field name and make it an int32

Reply:
Hint is a first in a field name... anything in spec is implicitly "desired state". 

DurationSeconds or something similar is fine, and document it may or may not be honored by the issuer.

For prior art, see the token expiration request field:

```
// ExpirationSeconds is the requested duration of validity of the request. The
// token issuer may return a token with a different validity duration so a
// client needs to check the 'expiration' field in a response.
// +optional
ExpirationSeconds *int64 `json:"expirationSeconds" protobuf:"varint,4,opt,name=expirationSeconds"`
```

Reply:
Also, document known cases where the issuer would not honor it:

1. old signer that's unaware of the field
2. signer whose configured max is shorter than the requested duration
3. signer whose configured min is longer than the requested duration